### ISSUE TITLE
Add history of delegating & memberships in profile page

### ISSUE THREAD

**Issue by danieljames-dj** [thewca/worldcubeassociation.org#8096]
<img width="1213" alt="Screenshot 2023-07-15 at 8 43 12 AM" src="https://github.com/thewca/worldcubeassociation.org/assets/9512698/87e4789e-7e71-49b9-96ae-0c950b32e448">

In the above section, on the right side of 'Results', 'Records' & 'Map', we will have two more additional sections: 'Delegating' and 'Memberships'.

In each section, it will have two sub-sections - 'Active roles' and 'Past roles'. In each row, it will show the role, start date and end date (for active roles, end date won't be shown, instead something like 'Since <start date>').

Please share any input on the same - like if you have any other suggestions or any improvements, etc.

**Comment by maxidragon** [comment]
Hi!

I would like to work on this issue

**Comment by gregorbg** [comment]
This is quite a complex issue that will be affected by refactors soon. I don't think it's worth investing the effort now if we're going to make major changes to the table structures for Delegates soon anyways

**Comment by maxidragon** [comment]
@gregorbg what about memberships in teams/committees?

**Comment by danieljames-dj** [comment]
I think we have to split this task into 3 subtasks:

- Restructure team_members table
- Implement membership history
- Implement delegate history

@gregorbg by 'refactors', I hope you meant the delegates table restructuring. In that case, can we go ahead with first two tasks above (as mentioned by @maxidragon as well) as those two tasks are independent of delegates table restructuring?

For restructuring of team_members table, we might need following changes:
- Change table name to memberships (because it has the committees, councils, etc as well)
- Replace `team_leader` and `team_senior_member` with `status` (similar to delegates table) which can have values like:
  - 'member', 'senior-member', 'leader', etc.
  - It can also have values like 'chair', 'vice-chair', 'secretary', etc to accommodate.
  - It can have values like 'translator-<language>' to show that they are translator, so that we can handle translators more easily. Also, we will be able to recognize translators in website, so that people will get motivated to contribute for translation.

It is worth noting that the team_members have banned members and probation delegates as well. Maybe we need to think something for them - like should we still have them in memberships table or do we need a new separate table for them.

@gregorbg do you agree with this change?
@maxidragon which change will you be interested to work on if you take it? 'Implement membership history' (which will basically be UI, and a bit of backend sometimes, but no change in DB structures) or 'Restructure team_members table' (it will have lots of changes along with DB structure change, as changes has to be made in places where they are used)? Will be happy if you work in both as well, I'll do my best to guide you in the same if you are new to WCA repo. (provided @gregorbg will approve to start the work on this)

**Comment by maxidragon** [comment]
@danieljames-dj I'm interested in working on 'Implement membership history', but I can also try to restructure team_members table. Do you have any suggestions on how to start with table restructuring?

**Comment by gregorbg** [comment]
Splitting the changeset up in a similar way to what Daniel suggested is fine. @maxidragon This is a big project and I do not feel comfortable passing this on to you under the cloak of GitHub anonymity. If you really want to work on this, we need to schedule a meeting to figure out the details, and I definitely won't have time for that before Worlds.

**Comment by maxidragon** [comment]
@gregorbg Speedcubing and programming are my two biggest passions, and I would love to combine them and contribute my skills to the software developed by WCA. I understand that you might not feel comfortable with the idea of me making changes to the database, but I would like to attempt it if you agree. I also think that a meeting would be necessary to discuss the details, and I can wait until you have more time. If you prefer someone else to handle the restructuring of the tables, that's okay too, I can focus on the UI.

**Comment by gregorbg** [comment]
The problem here is not that we're uncomfortable about you or your work, the problem is that we're a (mostly) volunteer team with limited resources and I simply don't have the time to think about design decisions like this during the "high season" of big competitions (US Nats and Worlds). You are free to reach out again towards the end of August, when we can discuss a more concrete implementation.

**Comment by danieljames-dj** [comment]
I've created [another issue](https://github.com/thewca/worldcubeassociation.org/issues/8122) to discuss and track memberships table along with delegates table. So this issue can be used to track just the UI part of history of delegating & memberships. @maxidragon feel free to comment on the design ideas in that issue. We can start the implementation of the same once the model is fixed after discussions.

**Comment by danieljames-dj** [comment]
Blocked by https://github.com/thewca/worldcubeassociation.org/issues/8122. But open to discuss how the UI should look like.
